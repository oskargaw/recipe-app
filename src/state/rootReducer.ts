import { combineReducers } from "redux";
import { recipesReducer } from "./recipes/recipes.reducer";

const rootReducer = combineReducers({
  recipes: recipesReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
