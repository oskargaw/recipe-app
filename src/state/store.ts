import { createStore } from "redux";
import { persistReducer, persistStore } from "redux-persist"; // imports from redux-persist
import storage from "redux-persist/lib/storage";
import rootReducer from "./rootReducer";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(
  persistedReducer,
  (window as any).__REDUX_DEVTOOLS_EXTENSION__ &&
    (window as any).__REDUX_DEVTOOLS_EXTENSION__()
);

const persistor = persistStore(store as any);

export { store, persistor };
