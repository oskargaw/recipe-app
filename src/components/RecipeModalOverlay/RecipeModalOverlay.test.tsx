import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { RootState } from "../../state/rootReducer";
import {
  RecipeModalOverlay,
  RecipeModalOverlayProps,
} from "./RecipeModalOverlay";

describe("<RecipeModalOverlay />", () => {
  const mockStore = configureStore([]);
  const initialState: RootState = {
    recipes: {
      recipeList: [],
    },
  };
  const store = mockStore(initialState);

  const mockOnClick = jest.fn();

  const props: RecipeModalOverlayProps = {
    children: "test",
    onClick: mockOnClick,
  };

  it("should render correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <RecipeModalOverlay {...props} />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
