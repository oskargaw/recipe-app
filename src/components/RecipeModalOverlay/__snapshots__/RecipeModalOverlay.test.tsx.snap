// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeModalOverlay /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL haQLmi"
    style="opacity: 0;"
  >
    test
  </div>
</DocumentFragment>
`;
